myapp/shared.py

```diff
@@ -0,0 +1,164 @@
+import spotipy
+from spotipy.oauth2 import SpotifyOAuth
+import pandas as pd
+from sklearn.preprocessing import MinMaxScaler
+
+# Spotify API credentials - you can use mine for testing
+SPOTIPY_CLIENT_ID = '4e6464520c2b4cf3a90ad792e5ad6504'
+SPOTIPY_CLIENT_SECRET = '4dde9d4fbb604ad3b22c0ad7204e25a0'
+SPOTIPY_REDIRECT_URI = 'http://localhost:8888/callback'
+SCOPE = 'user-top-read'
+
+# authentication
+sp = spotipy.Spotify(auth_manager=SpotifyOAuth(client_id=SPOTIPY_CLIENT_ID,
+                                               client_secret=SPOTIPY_CLIENT_SECRET,
+                                               redirect_uri=SPOTIPY_REDIRECT_URI,
+                                               scope=SCOPE))
+
+class GetTrack:
+    """
+    A class with functions that retrieve various information about tracks (for now: ID and features).
+    """
+
+    @staticmethod
+    def get_track_ids(tracks, is_playlist=False):
+        """
+        Returns a list of track IDs.
+
+        :param tracks: List of track objects
+        :type tracks: dict
+        :param is_playlist: Boolean indicating if the tracks are from a playlist
+        :type is_playlist: bool
+        :return: A list of track IDs
+        """
+        if is_playlist:
+            return [track['track']['id'] for track in tracks['items']]
+        else:
+            return [track['id'] for track in tracks['items']]
+
+    @staticmethod
+    def get_track_features(sp, track_ids):
+        """
+        Retrieves track information for multiple tracks in a batch request.
+        """
+        tracks_meta = sp.tracks(track_ids)
+        track_features = [
+            [
+                meta['name'],
+                meta['album']['name'],
+                meta['album']['artists'][0]['name'],
+                meta['external_urls']['spotify'],
+                meta['album']['images'][0]['url']
+            ]
+            for meta in tracks_meta['tracks']
+        ]
+        return track_features
+
+    @staticmethod
+    def get_audio_features(sp, track_ids):
+        """
+        Retrieves audio features for multiple tracks in a batch request.
```

Comment:
Could also be expanded on, so that for each function you get an explanation, and don't have to search through other function help files for certain explanations.